Engine_Revit_UI/Query/CeilingPattern.cs

```diff
@@ -0,0 +1,147 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+using Autodesk.Revit.DB;
+using BH.Engine.Geometry;
+using BH.oM.Geometry;
+
+namespace BH.UI.Revit.Engine
+{
+    public static partial class Query
+    {
+        public static List<BH.oM.Geometry.Line> CeilingPattern(this Ceiling ceiling, PlanarSurface surface)
+        {
+            CeilingType ceilingType = ceiling.Document.GetElement(ceiling.GetTypeId()) as CeilingType;
+            CompoundStructure comStruct = ceilingType.GetCompoundStructure();
+
+            if (comStruct != null && comStruct.GetLayers().Count > 0)
+                return (ceiling.Document.GetElement(comStruct.GetLayers()[0].MaterialId) as Material).CeilingPattern(surface);
+
+            List<ElementId> materialIds = ceiling.GetMaterialIds(false).ToList();
+            if (materialIds.Count == 0)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Ceiling patterns could not be pulled for ceiling element with the ID " + ceiling.Id);
+                return new List<oM.Geometry.Line>();
+            }
+
+            return (ceiling.Document.GetElement(materialIds[0]) as Material).CeilingPattern(surface);
+        }
+
+        public static List<BH.oM.Geometry.Line> CeilingPattern(this Material revitMaterial, PlanarSurface surface)
+        {
+            BoundingBox box = surface.IBounds();
+            double z = surface.ExternalBoundary.IControlPoints().Max(x => x.Z);
```

Comment:
why not `box.Max.Z`?

Reply:
The `BoundingBox` in BHoM is in real-world space, whereby the Z value (min or max) of the box might not be the Z level of the ceiling itself